### ISSUE TITLE
Mask not working after sub selection part 

### ISSUE THREAD

**Issue by szopqa** [nemtsov/json-mask#98]
Hello, 
I have encountered an issue processing following JSON:
```javascript
const exampleObject = {
    rootObjA: {
        nestedArrayA: [
            {
                firstPropToKeep: 123,
                secondPropToKeep: 321
            },
            {
                firstPropToKeep: 435,
                secondPropToKeep: 111
            }
        ]
    },
    rootObjB: {
        nestedFieldB: "some value i want to remove"
    },
    rootArrayA: [
        {
            nestedProp: 321
        },
        {
            nestedProp: 888
        }
    ]
}
```

Given following functionally equal masks:
```javascript
const workingMask = 'rootArrayA/*,rootObjA(nestedArrayA/*)';
const notWorkingMask = 'rootObjA(nestedArrayA/*),rootArrayA/*';
```
I expect following output for both of them:
```javascript
const expectedObj = {
    rootObjA: {
        nestedArrayA: [
            {
                firstPropToKeep: 123,
                secondPropToKeep: 321
            },
            {
                firstPropToKeep: 435,
                secondPropToKeep: 111
            }
        ]
    },
    rootArrayA: [
        {
            nestedProp: 321
        },
        {
            nestedProp: 888
        }
    ]
}
```

But for some reason I encounter two different results:
```javascript
const filteredObjWorking = mask(exampleObject, workingMask);
assert.deepEqual(filteredObjWorking, expectedObj); // fine
/**
{
  "rootArrayA": [
    {
      "nestedProp": 321
    },
    {
      "nestedProp": 888
    }
  ],
  "rootObjA": {
    "nestedArrayA": [
      {
        "firstPropToKeep": 123,
        "secondPropToKeep": 321
      },
      {
        "firstPropToKeep": 435,
        "secondPropToKeep": 111
      }
    ]
  }
}
 */

const filteredObjNotWorking = mask(exampleObject, notWorkingMask);
assert.deepEqual(filteredObjNotWorking, expectedObj); // error
/**
{
  "rootObjA": {
    "nestedArrayA": [
      {
        "firstPropToKeep": 123,
        "secondPropToKeep": 321
      },
      {
        "firstPropToKeep": 435,
        "secondPropToKeep": 111
      }
    ]
  }
}
*/
```

Could you please help me with that? It seems like a bug. The problem is that whatever is defined in mask after 'a(b,c) sub-selection' part for some reason is skipped. 
Regards

**Comment by alexrainman** [comment]
I have exactly the same problem.

Working:

```
fields=address(city,geo),company(name,bs)
```

Not working:

```
fields=address(city,geo/lat),company(name,bs)
```

So, adding /lat makes company to be ignored.

**Comment by alexrainman** [comment]
I found that inverting the query makes it work:

```
fields=company(name,bs),address(city,geo/lat)
```

**Comment by nemtsov** [comment]
This is resolved in **v1.0.4** thanks for reporting it @szopqa
